tests/test_pipeline.py

```diff
@@ -81,3 +88,29 @@ def update_state_dict(key: str):
     gc.collect()
     assert state_dict["source"]
     assert state_dict["sink"]
+
+
+@pytest.mark.use_cudf
+def test_pipeline_narrowing_types(config: Config, filter_probs_df: DataFrameType):
+    """
+    Test to ensure that we aren't narrowing the types of messages in the pipeline.
+
+    In this case, `ConvMsg` emits `MultiResponseMessage` messages which are a subclass of `MultiMessage`,
+    which is the accepted type for `MultiMessagePassThruStage`. We want to ensure that the type is retained allowing us
+    to place a stage after `MultiMessagePassThruStage` requring `MultiResponseMessage` like `AddScoresStage`.
+    """
+    config.class_labels = ['frogs', 'lizards', 'toads', 'turtles']
+    expected_df = filter_probs_df.to_pandas()
+    expected_df = expected_df.rename(columns=dict(zip(expected_df.columns, config.class_labels)))
+
+    pipe = LinearPipeline(config)
+    pipe.set_source(InMemorySourceStage(config, [filter_probs_df]))
+    pipe.add_stage(DeserializeStage(config))
+    pipe.add_stage(ConvMsg(config))
+    pipe.add_stage(MultiMessagePassThruStage(config))
+    pipe.add_stage(AddScoresStage(config))
```

Comment:
In this case it isn't needed as:
1) When I first wrote this test it failed do to an earlier bug in this refactor.
2) It will execute the test in C++ mode, if the types are somehow not as the C++ impls expect it will fail.